services/keybase/keybase-pgp.service.js

```diff
@@ -0,0 +1,106 @@
+'use strict'
+
+const KeybaseProfile = require('./keybase-profile')
+const Joi = require('joi')
+const { nonNegativeInteger } = require('../validators')
+
+const keyFingerprintSchema = Joi.object({
+  status: Joi.object({
+    code: nonNegativeInteger.required(),
+  }).required(),
+  them: Joi.array()
+    .items(
+      Joi.object({
+        public_keys: {
+          primary: {
+            key_fingerprint: Joi.string()
+              .hex()
+              .required(),
+          },
+        },
+      })
+        .required()
+        .allow(null)
+    )
+    .min(0)
+    .max(1),
+}).required()
+
+module.exports = class KeybasePGP extends KeybaseProfile {
+  static get apiVersion() {
+    return '1.0'
+  }
+
+  static get category() {
+    return 'social'
+  }
+
+  static get route() {
+    return {
+      base: 'keybase/pgp',
+      pattern: ':username',
+    }
+  }
+
+  static get defaultBadgeData() {
+    return {
+      label: 'pgp',
+      color: 'informational',
+    }
+  }
+
+  async handle({ username }) {
+    const options = {
+      form: {
+        usernames: username,
+        fields: 'public_keys',
+      },
+    }
+
+    const data = await this.fetch({
+      schema: keyFingerprintSchema,
+      options,
+    })
+
+    if (data.status.code !== 0) {
+      return {
+        message: 'invalid username',
+        color: 'critical',
+      }
+    }
+
```

Comment:
We've got these same two `if` blocks repeated in every service class. Let's refactor these into a `transform` method in the base class (I'll add more details later on)

Reply:
Good idea!